services/networking/loadbalancer/taxonomy.md

```diff
@@ -0,0 +1,67 @@
+| CCC-050322  | Connection draining (deregistration delay)            | Ability to gracefully terminate active connections to backend instances before they are removed from the load balancer's pool, to allow ongoing requests to complete successfully without disruption.                                                                                                                         |
+| CCC-050323  | Flow termination / Proxy behavior                     | Proxy incoming requests to the selected backend server or servers, establishing a new connection to the backend server and forwarding the request while ensuring the integrity of the original request is maintained.                                                                                                         |
+| CCC-050324  | URL redirects                                         | Redirect incoming traffic to a different URL or location.                                                                                                                                                                                                                                                                     |
+| CCC-050325  | URL rewrites                                          | Rewrite URL paths before forwarding them to backend services.                                                                                                                                                                                                                                                                 |
+| CCC-050326  | Configurable timeouts                                 | Configure timeout settings related to the handling of connections and requests by the load balancer. These settings may include idle connection timeouts, request processing timeouts, and health check timeouts                                                                                                              |
+| CCC-050327  | Custom error responses / fixed response               | Users have the capability to configure specific HTTP responses to be returned by the load balancer under defined conditions.                                                                                                                                                                                                  |
+| CCC-050328  | Outlier detection                                     | Configure the criteria for eviction of unhealthy backends, along with criteria defining when a backend or endpoint is considered healthy enough to receive traffic again.                                                                                                                                                     |
```

Comment:
I wonder if there's a difference and/or a crossover between this and 'health checks', and whether I should be more explicit with that.

Reply:
Do you mean the difference between a machine that's unhealthy (not responding to pings, for example) and a machine that's just slower than all the rest?

Reply:
I also thinks CCC-050328 and CCC-050351 are the same.

Reply:
> Do you mean the difference between a machine that's unhealthy (not responding to pings, for example) and a machine that's just slower than all the rest?

I think that's a good example of the slight difference. I think to me 'outlier detection' works alongside/underneath 'health checks'. Health checks is the basic check of whether or not a backend can send a valid response within a set time. Outlier detection identifies and isolates backends that are 'behaving unusually'- so monitoring metrics to assess performance of all machines and using that as a baseline to identify those that are deviating significantly- and adjusts the traffic it sends to it accordingly.

However, that's my interpretation, and it's possible that a) I'm misinterpreting and b) the distinctions aren't significant enough to require two separate features in this taxonomy.

Reply:
I think your interpretation is good and is in good agreement with my own understanding!